### PR TITLE
feat: add consensys zkevm (Linea) support

### PR DESCRIPTION
**Pull request by VGau**
=======

**Description**
Add support for ConsenSys zkEVM network (Linea).

**Screenshots/Recordings**
Screenshot 2023-03-15 at 2 00 32 PM

https://user-images.githubusercontent.com/85494462/227060203-6dbee5d6-09ca-4adc-9726-7e3164fc21e9.mp4

**Acceptance Criteria**
## Manual Testing Steps

Step 1:
Go to this website: https://bridge.goerli.zkevm.consensys.net/send?token=ETH and send some ETH (minimum 0.1) from Goerli to Consensys zkEVM (Linea) .
Screenshot 2023-02-23 at 12 04 31 PM

Wait for transaction to be finished.

Step 2:
Open the metamask mobile app and select then new ConsenSys zkEVM (Linea) test network in the dropdown.
Screenshot 2023-03-15 at 2 00 32 PM

Step 3:
Go to this website: https://bridge.goerli.zkevm.consensys.net/send?token=ETH and send some ETH (minimum 0.1) from  Consensys zkEVM (Linea) to Goerli.
Screenshot 2023-02-23 at 12 07 30 PM

Wait for transaction to be finished.

Final step:
Check if all transactions succeed on Metamask.
- What Dapps does it support?

**Related Tickets**
Core PR (https://github.com/MetaMask/core/pull/1126) to follow 

**Checklist**

* [X] There is a related GitHub issue
* [X] Tests are included if applicable
* [X] Any added code is fully documented

### CODE REVIEW

**Comment by VGau** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by sethkfman — COMMENTED** [review]
One minor improvement suggested

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]
This PR is 🌮 🌮 🌮

File: app/components/UI/NetworkList/index.js [hunk 1]
```diff
@@ -440,6 +531,13 @@ export class NetworkList extends PureComponent {
           {this.renderMainnet()}
           {this.renderRpcNetworks()}
           {this.renderOtherNetworks()}
+          {shouldShowZKEVM &&
+            !isLineaTestnetInFrequentRpcList &&
+            this.renderNonInfuraNetwork(
+              NETWORKS_CHAIN_ID.LINEA_TESTNET,
+              LINEA_TESTNET_RPC_URL,
+              'Linea Goerli Test Network',
```
Comment (sethkfman) [inline app/components/UI/NetworkList/index.js:539]: This needs to be added to our locales/en.js file and referenced using `strings()`

File: app/constants/network.js [hunk 2]
```diff
@@ -8,6 +8,11 @@ export const RPC = 'rpc';
 export const NO_RPC_BLOCK_EXPLORER = 'NO_BLOCK_EXPLORER';
 export const PRIVATENETWORK = 'PRIVATENETWORK';
 export const DEFAULT_MAINNET_CUSTOM_NAME = 'Ethereum Main Custom';
+export const LINEA_TESTNET_BLOCK_EXPLORER =
+  'https://explorer.goerli.linea.build';
+export const LINEA_TESTNET_RPC_URL = 'https://rpc.goerli.linea.build';
```
Comment (sethkfman) [inline app/constants/network.js:13]: These URL should go into constants/url file

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**